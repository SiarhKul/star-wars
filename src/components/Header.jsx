import React from 'react';
import { Link, NavLink } from 'react-router-dom';

export const Header = () => {
  return (
    <header className="header">
      <div className="logo"> Star Wars</div>
      <nav className="nav">
        <ul className="nav-list">
          <li className="nav-list__item">
            <NavLink
              className="nav-list__link"
              activeClassName="selected"
              to="/"
              exact
            >
              People
            </NavLink>
          </li>
          <li className="nav-list__item">
            <NavLink
              activeClassName="selected"
              className="nav-list__link"
              to="/planets"
            >
              Planets
            </NavLink>
          </li>
          <li className="nav-list__item">
            <NavLink
              activeClassName="selected"
              className="nav-list__link"
              to="/starships"
            >
              Starship
            </NavLink>
          </li>
          {/* <li className="nav-list__item">People</li> */}
          {/* <li className="nav-list__item">Planets</li>
          <li className="nav-list__item">Starships</li> */}
        </ul>
      </nav>
    </header>
  );
};
