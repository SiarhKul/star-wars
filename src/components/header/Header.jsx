import React from "react";
import { NavLink } from "react-router-dom";

export const Header = () => {
	const menuItems = [
		"People",
		"Planets",
		"Starships",
		"Films",
		"Vehicles",
		"Species",
	];

	return (
		<header className="header">
			<div className="logo"> Star Wars</div>
			<nav className="nav">
				<ul className="nav-list">
					{menuItems.map(item => {
						return (
							<li key={item} className="nav-list__item">
								<NavLink
									className="nav-list__link"
									activeClassName="selected"
									to={item === "People" ? "/" : item.toLowerCase()}
									exact
								>
									{item}
								</NavLink>
							</li>
						);
					})}
				</ul>
			</nav>
		</header>
	);
};
