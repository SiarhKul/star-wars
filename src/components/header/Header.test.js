import React from "react";
import { shallow } from "enzyme";
import { Header } from "./Header";

const header = () => shallow(<Header />);

describe("Test 'Header' component", () => {
	let component;

	beforeEach(() => {
		component = header();
	});

	it("should component render", () => {
		expect(component).toMatchSnapshot();
	});

	it("should component has 6 navlinks ", () => {
		const wrapper = component.find("NavLink");
		expect(wrapper).toHaveLength(7);
	});
});
