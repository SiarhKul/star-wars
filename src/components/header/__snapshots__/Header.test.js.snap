// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 'Header' component should component render 1`] = `
<header
  className="header"
>
  <div
    className="logo"
  >
     Star Wars
  </div>
  <nav
    className="nav"
  >
    <ul
      className="nav-list"
    >
      <li
        className="nav-list__item"
      >
        <NavLink
          className="nav-list__link"
          to="/"
        >
          <FaHome />
        </NavLink>
      </li>
      <li
        className="nav-list__item"
        key="People"
      >
        <NavLink
          activeClassName="selected"
          className="nav-list__link"
          exact={true}
          to="people"
        >
          People
        </NavLink>
      </li>
      <li
        className="nav-list__item"
        key="Planets"
      >
        <NavLink
          activeClassName="selected"
          className="nav-list__link"
          exact={true}
          to="planets"
        >
          Planets
        </NavLink>
      </li>
      <li
        className="nav-list__item"
        key="Starships"
      >
        <NavLink
          activeClassName="selected"
          className="nav-list__link"
          exact={true}
          to="starships"
        >
          Starships
        </NavLink>
      </li>
      <li
        className="nav-list__item"
        key="Films"
      >
        <NavLink
          activeClassName="selected"
          className="nav-list__link"
          exact={true}
          to="films"
        >
          Films
        </NavLink>
      </li>
      <li
        className="nav-list__item"
        key="Vehicles"
      >
        <NavLink
          activeClassName="selected"
          className="nav-list__link"
          exact={true}
          to="vehicles"
        >
          Vehicles
        </NavLink>
      </li>
      <li
        className="nav-list__item"
        key="Species"
      >
        <NavLink
          activeClassName="selected"
          className="nav-list__link"
          exact={true}
          to="species"
        >
          Species
        </NavLink>
      </li>
    </ul>
  </nav>
</header>
`;
